merlin/metaanalysis/aggregatedata.py

```diff
@@ -0,0 +1,55 @@
+from merlin.core import analysistask
+import pandas
+
+
+class AggregateData(analysistask.AnalysisTask):
+    """
+    A metaanalysis task that aggregates data from multiple datasets
+    """
+    def __init__(self, metaDataSet, parameters=None, analysisName=None):
+        super().__init__(metaDataSet, parameters, analysisName)
+
+        if 'overwrite' not in self.parameters:
+            self.parameters['overwrite'] = False
+
+        self.metaDataSet = metaDataSet
+
+    def get_estimated_memory(self):
+        return 10000
+
+    def get_estimated_time(self):
+        return 100
+
+    def get_dependencies(self):
+        dep = [v for k, v in self.parameters.items() if 'task' in k]
```

Comment:
This doesn't seem like it will work. We may need an AnalysisDependency class that can keep track of the full dependency, including both the analysis task name and the dataset and then update all the get_dependencies functions to return a list of AnalysisDependencies.

Reply:
Another option is to change all the analysis tasks to return their dependencies as AnalysisTask objects. For this, we have to make sure that it never tries to create an AnalysisTask object before the analysis task has actually been created.